src/main/java/net/panda/mekanismfabric/MekanismDataGenerator.java

```diff
@@ -2,10 +2,224 @@
+			ItemConvertible input = MekanismItems.PROCESSED_RESOURCES.get(ResourceTypes.INGOT, resource);
+			ItemConvertible output = MekanismItems.PROCESSED_RESOURCES.get(ResourceTypes.NUGGET, resource);
+
+			if (output != null && input != null) {
+				createShapelessIngotToNuggetRecipe(exporter, input, output);
+			}
+		}
+
+		private void createShapedIngotToBlockRecipeFromResource(Consumer<RecipeJsonProvider> exporter, PrimaryResources resource) {
+			ItemConvertible inputRaw = MekanismItems.PROCESSED_RESOURCES.get(ResourceTypes.RAW, resource);
+			ItemConvertible input = MekanismItems.PROCESSED_RESOURCES.get(ResourceTypes.INGOT, resource);
+			ItemConvertible outputRaw = MekanismBlocks.PROCESSED_RESOURCE_BLOCKS.get(resource.getRawResourceBlockInfo());
+			ItemConvertible output = MekanismBlocks.PROCESSED_RESOURCE_BLOCKS.get(resource);
+
+			if (input != null) {
+				createShapedIngotToBlockRecipe(exporter, input, output);
+			}
+
+			if (inputRaw != null) {
+				createShapedIngotToBlockRecipe(exporter, inputRaw, outputRaw);
+			}
+		}
+
+		private void createShapedNuggetToIngotRecipeFromResource(Consumer<RecipeJsonProvider> exporter, PrimaryResources resource) {
+			ItemConvertible input = MekanismItems.PROCESSED_RESOURCES.get(ResourceTypes.NUGGET, resource);
+			ItemConvertible output = MekanismItems.PROCESSED_RESOURCES.get(ResourceTypes.INGOT, resource);
+
+			if (output != null && input != null) {
+				createShapedNuggetToIngotRecipe(exporter, input, output);
+			}
+		}
+
+		private void createCookingRecipe(Consumer<RecipeJsonProvider> exporter, List<ItemConvertible> inputs, ItemConvertible output, String path) {
+			RecipeProvider.offerSmelting(exporter, inputs, RecipeCategory.MISC, output, 0.45F, 300, SMELTING_PATH);
+
+//			CookingRecipeJsonBuilder.createSmelting(ingredient, RecipeCategory.MISC, output, 0.45F, 300)
+//					.criterion(FabricRecipeProvider.hasItem(),
+//							FabricRecipeProvider.conditionsFromItem(input))
+//					.offerTo(exporter, SMELTING_PATH + path + extractIdentifier(output));
+//
+//			CookingRecipeJsonBuilder.createSmelting(ingredient, RecipeCategory.MISC, output, 0.45F, 300)
+//					.criterion(FabricRecipeProvider.hasItem(input),
+//							FabricRecipeProvider.conditionsFromItem(input))
+//					.offerTo(exporter, BLASTING_PATH + path + extractIdentifier(output));
+		}
+
+		private void createCookingRecipe(Consumer<RecipeJsonProvider> exporter, ItemConvertible input, ItemConvertible output, String path) {
+			createCookingRecipe(exporter, List.of(input), output, path);
+		}
+
+		@Override
+		public void generate(Consumer<RecipeJsonProvider> exporter) {
+			// Block to Ingot
+			createShapelessResourceBlockToIngotRecipe(exporter, PrimaryResources.OSMIUM);
+			createShapelessResourceBlockToIngotRecipe(exporter, PrimaryResources.TIN);
+			createShapelessResourceBlockToIngotRecipe(exporter, PrimaryResources.LEAD);
+			createShapelessResourceBlockToIngotRecipe(exporter, PrimaryResources.URANIUM);
```

Comment:
Loop over `PrimaryResources` and call
- `createShapelessResourceBlockToIngotRecipe`
- `createShapelessIngotToNuggetRecipeFromResource`
- `createShapedIngotBlockRecipeFromResource`
- `createShapedNuggetToIngotRecipeFromResource`